internal/pkg/docker/dockercompose/svc_image.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockercompose
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	compose "github.com/compose-spec/compose-go/types"
+	"github.com/dustin/go-humanize/english"
+)
+
+// convertImageConfig takes in a Compose build config & image field and returns a Copilot image manifest.
+func convertImageConfig(build *compose.BuildConfig, labels map[string]string, imageLoc string) (manifest.Image, IgnoredKeys, error) {
+	image := manifest.Image{
+		DockerLabels: labels,
+	}
+
+	// Compose allows both build & image to be specified, but image takes precedence.
+	// In Copilot, however, build and image are mutually exclusive.
+	if imageLoc != "" {
+		image.Location = aws.String(imageLoc)
+		return image, nil, nil
+	} else if build == nil {
+		return manifest.Image{}, nil, errors.New("missing one of `build` or `image`")
+	}
+
+	ignored, err := unsupportedBuildKeys(build)
+	if err != nil {
+		return manifest.Image{}, ignored, err
+	}
+
+	buildArgs := manifest.DockerBuildArgs{
+		Context:    nilIfEmpty(build.Context),
+		Dockerfile: nilIfEmpty(build.Dockerfile),
+		Target:     nilIfEmpty(build.Target),
+		CacheFrom:  build.CacheFrom,
+	}
+
+	args, err := convertMappingWithEquals(build.Args)
+	if err != nil {
+		return manifest.Image{}, nil, fmt.Errorf("convert build args: %w", err)
+	}
+
+	if len(args) != 0 {
+		buildArgs.Args = args
+	}
+
+	image.Build = manifest.BuildArgsOrString{
+		BuildArgs: buildArgs,
+	}
+
+	return image, ignored, nil
+}
+
+func nilIfEmpty(s string) *string {
+	if s == "" {
+		return nil
+	} else {
+		return &s
+	}
+}
+
+// convertMappingWithEquals checks for entries with missing values and generates an error if any are found.
+func convertMappingWithEquals(inArgs compose.MappingWithEquals) (map[string]string, error) {
+	args := map[string]string{}
+	var badArgs []string
+
+	for k, v := range inArgs {
+		if v == nil {
+			badArgs = append(badArgs, k)
+			continue
+		}
+		args[k] = *v
+	}
+
+	if len(badArgs) != 0 {
+		return nil, fmt.Errorf("%s '%v' %s missing %s and %s user input, this is unsupported in Copilot",
```

Comment:
What do you think about this (plus deleting "requires"/"require"? We get across the idea that this is not allowed in Copilot (even if it is in Docker Compose), but is less redundant.
```suggestion
		return nil, fmt.Errorf("%s '%v' %s missing %s; this is unsupported in Copilot",
```